modular_lumos/code/game/machinery/fish_tanks.dm

```diff
@@ -0,0 +1,227 @@
+/obj/structure/fish_tank_base
+	name = "unfinished fish tank"
+	desc = "A tank that requires some glass"
+	icon = 'modular_lumos/icons/obj/fish_items.dmi'
+	icon_state = "tanki1"
+
+	density = TRUE
+
+/obj/structure/fish_tank_base/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/stack/sheet/glass))
+		var/obj/item/stack/sheet/glass/glass = I
+		if(glass.amount < 5)
+			return
+		if(!do_after(user, 2 SECONDS, FALSE, src))
+			return
+		glass.use(5)
+		new /obj/structure/fish_tank(get_turf(src))
+		qdel(src)
+		return
+	else if(istype(I, /obj/item/screwdriver))
+		I.play_tool_sound(src, 50)
+		if(!do_after(user, 2 SECONDS, FALSE, src))
+			return
+		I.play_tool_sound(src, 50)
+		var/obj/item/stack/sheet/metal/metal = new /obj/item/stack/sheet/metal(get_turf(src))
+		metal.amount = 20
+		qdel(src)
+		return
+	else
+		return ..()
+
+/obj/structure/fish_tank
+	name = "fish tank"
+	desc = "A tank that is for growing fish."
+	icon = 'modular_lumos/icons/obj/fish_items.dmi'
+	icon_state = "tank1"
+
+	///Dirty: how dirty a tank is
+	var/dirty = 0
+	var/clean = 255
+
+	var/seaweed_grow = 0
+
+	var/current_food = 0
+
+	var/resting_period
+
+	density = TRUE
+
+/obj/structure/fish_tank/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
+	update_icon()
+
+/obj/structure/fish_tank/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/structure/fish_tank/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/fishy))
+		var/obj/item/fishy/fish = I
+		if(contents.len >= 2)
+			to_chat(user, "<span class='warning'>You cannot fit anymore fish!</span>")
+			return
+		fish.forceMove(src)
+		fish.in_tank = TRUE
+		update_icon()
+		return
+	else if(istype(I, /obj/item/fish_tool/brush))
+		if(!do_after(user, 2 SECONDS, FALSE, src))
+			return
+		dirty = 0
+		clean = 255
+		update_icon()
+		return
+	else if(istype(I, /obj/item/fish_tool/clippers))
+		if(seaweed_grow >= 50)
+			seaweed_grow -= 50
+			new /obj/item/reagent_containers/food/snacks/sea_weed(get_turf(src))
+		update_icon()
+		return
+	else if(istype(I, /obj/item/wirecutters))
+		if(seaweed_grow >= 50)
+			seaweed_grow -= 50
+			if(prob(50))
+				new /obj/item/reagent_containers/food/snacks/sea_weed(get_turf(src))
+		update_icon()
+		return
+	else if(istype(I, /obj/item/wrench))
+		I.play_tool_sound(src, 50)
+		if(!do_after(user, 2 SECONDS, FALSE, src))
+			return
+		I.play_tool_sound(src, 50)
+		anchored = !anchored
+		return
+	else if(istype(I, /obj/item/crowbar))
+		I.play_tool_sound(src, 50)
+		if(!do_after(user, 2 SECONDS, FALSE, src))
+			return
+		I.play_tool_sound(src, 50)
+		new /obj/structure/fish_tank_base(get_turf(src))
+		var/obj/item/stack/sheet/glass/glass = new /obj/item/stack/sheet/glass(get_turf(src))
+		glass.amount = 5
+		qdel(src)
+		return
+	else if(istype(I, /obj/item/fish_tool/fish_food))
+		var/obj/item/fish_tool/fish_food/food = I
+		if(food.food_left >= 5)
+			food.food_left -= 5
+			current_food += 5
```

Comment:
```suggestion
			current_food += 10
```
I feel that fish food is consumed too fast. I think doubling the quantity inserted would be a good idea.

Reply:
fixed/done